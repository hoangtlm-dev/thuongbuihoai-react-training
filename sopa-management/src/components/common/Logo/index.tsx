// Libs
import React, { memo } from 'react';
import { Link } from 'react-router-dom';

interface LogoProps {
  url: string;
  alt?: string;
}

const Logo: React.FC<LogoProps> = ({
  url,
  alt = ''
}) => (
  <Link data-testid='logo' to='/'>
    <img className='logo' src={url} alt={alt} />
  </Link>
);

export default memo(Logo);
