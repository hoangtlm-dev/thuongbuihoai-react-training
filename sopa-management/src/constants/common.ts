const SIZE = [
  {
    key: "1",
    label: "Small",
  },
  {
    key: "2",
    label: "Medium",
  },
  {
    key: "3",
    label: "Large",
  }
];

const USER = [
  {
    key: "1",
    label: "Logout"
  },
];

export {
  SIZE,
  USER,
};
