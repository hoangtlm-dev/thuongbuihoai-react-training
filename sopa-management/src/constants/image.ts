export const IMAGE = {
  whiteLogo: "/images/logos/white-logo.svg",
  blackLogo: "/images/logos/black-logo.svg",
  atl: "This is logo sopa",
};
