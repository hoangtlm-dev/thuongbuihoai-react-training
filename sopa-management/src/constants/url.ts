const BASE_URL = process.env.VITE_BASE_URL;
const PRODUCT_URL = `${BASE_URL}/products`;
const ACCOUNT_URL = `${BASE_URL}/accounts`;

export {
  PRODUCT_URL,
  ACCOUNT_URL
};
