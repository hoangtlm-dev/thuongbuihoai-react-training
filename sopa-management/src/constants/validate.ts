const CONFIRM_MESSAGE = {
  ADD_SUCCESS: 'Add product to cart successfully',
  DELETE_SUCCESS: 'Delete product to cart successfully',
  CONFIRM_DELETE: 'Are you sure you want to delete?'
}

const ERROR_MESSAGES = {
  REQUIRED: 'This field is required',
  EMAIL_FORMAT: 'Wrong email format',
  EMAIL_INCORRECT: 'Email is incorrect',
  PASSWORD_FORMAT: 'Password must be more than 6 characters',
  PASSWORD_INCORRECT: 'Password is incorrect',
}

export {
  CONFIRM_MESSAGE,
  ERROR_MESSAGES,
};
