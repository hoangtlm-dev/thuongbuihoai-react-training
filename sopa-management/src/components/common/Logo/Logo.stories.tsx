import type { Meta, StoryObj } from '@storybook/react';
import Logo from '.';
import { IMAGE } from '../../../constants/image';

const meta: Meta<typeof Logo> = {
  component: Logo,
};

export default meta;
type Story = StoryObj<typeof Logo>;

export const WhiteLogo: Story = {
  args: {
    url: IMAGE.whiteLogo,
    alt: IMAGE.atl,
  }
};

export const BlackLogo: Story = {
  args: {
    url: IMAGE.blackLogo,
    alt: IMAGE.atl,
  }
};

