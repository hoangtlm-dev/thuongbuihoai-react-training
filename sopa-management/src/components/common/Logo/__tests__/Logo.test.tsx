import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { IMAGE } from "../../../../constants/image";
import Logo from "..";

describe("Logo component", () => {
  test("should render logo component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <Logo url={IMAGE.blackLogo} />
      </Router>
    );
    expect(getByTestId("logo")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <Logo url={IMAGE.whiteLogo} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
