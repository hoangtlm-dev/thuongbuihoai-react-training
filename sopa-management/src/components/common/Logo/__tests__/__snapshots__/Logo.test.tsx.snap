// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component matches snapshot 1`] = `
<DocumentFragment>
  <a
    data-testid="logo"
    href="/"
  >
    <img
      class="logo"
      src="/public/images/logos/white-logo.svg"
    />
  </a>
</DocumentFragment>
`;
